airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/utils/dagster_helpers.py

```diff
@@ -1,12 +1,24 @@
 from dagster import MetadataValue, Output
 import pandas as pd
-from typing import NewType
+from typing import Optional, List
 
 OutputDataFrame = Output[pd.DataFrame]
+CURSOR_SEPARATOR = ":"
 
 
 def output_dataframe(result_df: pd.DataFrame) -> Output[pd.DataFrame]:
     """
     Returns a Dagster Output object with a dataframe as the result and a markdown preview.
     """
     return Output(result_df, metadata={"count": len(result_df), "preview": MetadataValue.md(result_df.to_markdown())})
+
+
+def deserialize_composite_etag_cursor(etag_cursor: Optional[str]) -> List[str]:
+    if etag_cursor is None:
+        return []
+
+    return etag_cursor.split(CURSOR_SEPARATOR)
+
+
+def serialize_composite_etag_cursor(etags: List[str]):
```

Comment:
Will do. And it is because they only support string cursors at this time
https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#idempotence-and-cursors